apps/ensindexer/src/lib/plugin-helpers.ts

```
@@ -167,14 +156,25 @@ export const activateHandlers =
 /**
  * Defines a ponder#NetworksConfig for a single, specific chain.
  */
-export function networksConfigForChain(chain: Chain) {
+export function networksConfigForChain(config: ENSIndexerConfig, chainId: number) {
   return {
-    [chain.id.toString()]: {
-      chainId: chain.id,
-      transport: http(rpcEndpointUrl(chain.id)),
-      maxRequestsPerSecond: rpcMaxRequestsPerSecond(chain.id),
-      // NOTE: disable cache on 'Anvil' chains
-      ...(chain.name === "Anvil" && { disableCache: true }),
+    [chainId.toString()]: {
+      chainId: chainId,
+      // This may return undefined if the RPC URL is not configured for the chain.
+      // This is intentional to allow us to aggregate all the RPC URLs that are missing in a
+      // later validation step allowing us to throw a helpful error message aggregating them
+      // all instead of throwing an error here which will only flag a single missing RPC URL.
+      // The code which does that validation is the `validateChainConfigs` function in
+      // `src/config/validations.ts`.
+      transport: http(config.indexedChains[chainId]?.rpcEndpointUrl),
```

This function should first validate that `config.indexedChains[chainId]` is defined. If it isn't, it should throw an exception.

After validating that, there should be no need for the `?` here.